src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -180,6 +181,38 @@ public interface OnHitEffect
 		}
 	};
 	
+	/**
+	 * A knockback effect that does a right-vector (relative to the player's orientation) and direction to enemy dot calculation (relative to the player's orientation) to spread enemies horizontally via knockback.
+	 */
+	OnHitEffect SPREADING_KNOCKBACK = (stack, target, attacker) -> {
+		
+		//Attacker's right direction (forward value rotated 90°)
+		Vec3 attackerRight = new Vec3(
+				(float) (Math.cos(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0)),
+				0,
+				(float) (Math.sin(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0))
+		);
+		
+		//Direction vector (attacker -> target)
+		Vec3 dirToTarget = attacker.position().multiply(1,0,1)
+				.vectorTo(target.position().multiply(1,0,1));
+		dirToTarget = dirToTarget.normalize();
+		
+		//Dot product of direction and attacker's right
+		double dot = attackerRight.dot(dirToTarget);
+		double expDot = 1.5 * Math.pow(1 - (Math.abs(dot)), 0.3);
+		
+		//Knockback direction vector
+		Vec3 knockback = new Vec3(
+				(float) (-dirToTarget.x + (attackerRight.x * Mth.sign(dot) * expDot)),
+				0,
+				(float) (-dirToTarget.z + (attackerRight.z * Mth.sign(dot) * expDot)));
```

Comment:
```suggestion
		Vec3 knockback = dirToTarget.scale(-1).add(attackerRight.scale(Mth.sign(dot) * expDot));
```
And consider replacing `Mth.sign(dot) * expDot` using:
```java
double sideKnockbackFactor = Mth.sign(dot) * 1.5 * Math.pow(1 - (Math.abs(dot)), 0.3);
```

Reply:
Applied, as well as some other likewise simplifications.